src/analysis/normed/group/seminorm.lean

```diff
@@ -52,6 +56,16 @@ structure group_seminorm (G : Type*) [group G] :=
 (mul_le' : ∀ x y, to_fun (x * y) ≤ to_fun x + to_fun y)
 (inv' : ∀ x, to_fun x⁻¹ = to_fun x)
 
+/-- A nonarchimedean seminorm on an additive group `G` is a function `f : G → ℝ` that preserves
+zero, is nonarchimedean and such that `f (-x) = f x` for all `x`. -/
+structure nonarch_add_group_seminorm (G : Type*) [add_group G] extends zero_hom G ℝ :=
```

Comment:
```suggestion
@[protect_projs]
structure nonarch_add_group_seminorm (G : Type*) [add_group G] extends zero_hom G ℝ :=
```

Reply:
Seems like you forgot this one

Reply:
I skipped this one because adding `@[protect_proj]` here causes 42 errors later in the file (starting with an `unknown identifier 'to_fun'` in line 351.

Reply:
Do you mean 
```
noncomputable instance : has_coe_to_fun (nonarch_add_group_seminorm E) (λ _, E → ℝ) :=
fun_like.has_coe_to_fun
```
and then adding `noncomputable` where it complains?

I have also noticed that there is no `protect_proj` in `add_group_seminorm`, should I add it there too? Otherwise, could you explain why the two structures are treated differently?

Reply:
I see you understood what I meant 😄 

Tagging `add_group_seminorm` with `protect_projs` is perfect.